FileIO/XmlIO/Boost/BoostXmlGmlInterface.cpp

```diff
@@ -58,222 +81,148 @@ bool BoostXmlGmlInterface::readFile(const std::string &fname)
 	{
-		if (point.first.compare("point") != 0)
-			continue;
-
-		unsigned p_id = static_cast<unsigned>(point.second.get("<xmlattr>.id", std::numeric_limits<unsigned>::max()));
-		double   p_x  = static_cast<double>  (point.second.get("<xmlattr>.x",  std::numeric_limits<double>::max()));
-		double   p_y  = static_cast<double>  (point.second.get("<xmlattr>.y",  std::numeric_limits<double>::max()));
-		double   p_z  = static_cast<double>  (point.second.get("<xmlattr>.z",  std::numeric_limits<double>::max()));
-		std::string p_name = point.second.get("<xmlattr>.name", "");
-
-		if ( p_id == std::numeric_limits<unsigned>::max() || p_x == std::numeric_limits<double>::max() ||
-		     p_y  == std::numeric_limits<double>::max()   || p_z == std::numeric_limits<double>::max() )
-			WARN("BoostXmlGmlInterface::readPoints(): Skipping point, attribute missing in <point> tag:\n%s",
-				BaseLib::propertyTreeToString(point.second).c_str())
-		else
+		auto const p_id = pt.getConfAttribute<std::size_t>("id");
+		auto const p_x  = pt.getConfAttribute<double>("x");
+		auto const p_y  = pt.getConfAttribute<double>("y");
+		auto const p_z  = pt.getConfAttribute<double>("z");
+
+		auto const p_size = points.size();
+		insert_if_key_unique_else_error(_idx_map, p_id, p_size,
+		    "The point id is not unique.");
+		points.push_back(new GeoLib::Point(p_x, p_y, p_z, p_id));
+
+		if (auto const p_name = pt.getConfAttributeOptional<std::string>("name"))
 		{
-			_idx_map.insert (std::pair<std::size_t, std::size_t>(p_id, points->size()));
-			GeoLib::Point* p = new GeoLib::Point(p_x, p_y, p_z, p_id);
-			if (!p_name.empty())
-				pnt_names->insert( std::pair<std::string, std::size_t>(p_name, points->size()) );
-			points->push_back(p);
-		}
-	}
+			if (p_name->empty()) {
+				ERR("Empty point name found in geometry file.");
+				std::abort();
+			}
 
-	// if names-map is empty, set it to nullptr because it is not needed
-	if (pnt_names->empty())
-	{
-		delete pnt_names;
-		pnt_names = nullptr;
+			insert_if_key_unique_else_error(pnt_names, *p_name, p_size,
+			    "The point name is not unique.");
+		}
 	}
 }
 
 void BoostXmlGmlInterface::readPolylines(
-    BaseLib::ConfigTree const& polylinesRoot,
-    std::vector<GeoLib::Polyline*>* polylines,
-    std::vector<GeoLib::Point*> const* points,
-    const std::vector<std::size_t>& pnt_id_map,
-    std::map<std::string, std::size_t>*& ply_names)
+    BaseLib::ConfigTreeNew const& polylinesRoot,
+    std::vector<GeoLib::Polyline*>& polylines,
+    std::vector<GeoLib::Point*> const& points,
+    std::vector<std::size_t> const& pnt_id_map,
+    std::map<std::string, std::size_t>& ply_names)
 {
-	BOOST_FOREACH( BaseLib::ConfigTree::value_type const & polyline, polylinesRoot )
+	for (auto const pl : polylinesRoot.getConfSubtreeList("polyline"))
 	{
-		if (polyline.first.compare("polyline") != 0)
-			continue;
+		auto const id = pl.getConfAttribute<std::size_t>("id");
+		(void) id; // id not used
```

Comment:
TODO for myself: The id is not used but it must be there in the gml file. That's why `ignore...()` cannot be used.